### PR TITLE
NYP: added Pakistani holidays and test case

### PR DESCRIPTION
**Pull request by NabeelYousafPasha**
- Added all Pakistani public holidays
- Left Islamic Holidays as they are lunar and region bases

### CODE REVIEW

File: src/Countries/Pakistan.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Pakistan extends Country
+{
+    /**
+     * Country Code
```
Comment (Nielsvanpach) [inline src/Countries/Pakistan.php:10]: this can be removed

File: src/Countries/Pakistan.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     *
+     * @return string
```
Comment (Nielsvanpach) [inline src/Countries/Pakistan.php:12]: this can be removed

File: src/Countries/Pakistan.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function countryCode(): string
+    {
+        return 'pk';
+    }
+
+    /**
+     * All Holiday In Country
```
Comment (Nielsvanpach) [inline src/Countries/Pakistan.php:20]: You can drop these docblocks too

**Review by Nielsvanpach — COMMENTED** [review]
Thanks! Did you verify if the generated results in the snapshot are correct?

**Comment by muhammad-nauman** [comment]
see #19

File: src/Countries/Pakistan.php (hunk 1 above)
Comment (NabeelYousafPasha) [reply src/Countries/Pakistan.php]: Removed

File: src/Countries/Pakistan.php (hunk 2 above)
Comment (NabeelYousafPasha) [reply src/Countries/Pakistan.php]: Removed

File: src/Countries/Pakistan.php (hunk 3 above)
Comment (NabeelYousafPasha) [reply src/Countries/Pakistan.php]: Removed

**Comment by NabeelYousafPasha** [comment]
> Thanks! Did you verify if the generated results in the snapshot are correct?

Yes, I verified those.
Snapshot content attached as well.

`[
    {
        "name": "Kashmir Solidarity Day",
        "date": "2024-02-05"
    },
    {
        "name": "Pakistan Resolution Day",
        "date": "2024-03-23"
    },
    {
        "name": "Labour Day",
        "date": "2024-05-01"
    },
    {
        "name": "Independence Day",
        "date": "2024-08-14"
    },
    {
        "name": "Defence Day",
        "date": "2024-09-06"
    },
    {
        "name": "Iqbal Day",
        "date": "2024-11-09"
    },
    {
        "name": "Quaid-e-Azam Day",
        "date": "2024-12-25"
    }
]`

**Comment by muhammad-nauman** [comment]
@NabeelYousafPasha can you please close this in favor of #19 as per the guidelines?
Thanks!

**Comment by NabeelYousafPasha** [comment]
Closing this PR as https://github.com/spatie/holidays/pull/19 covers Pakistan Holidays.

Thanks @muhammad-nauman for your contribution

**Comment by muhammad-nauman** [comment]
Thanks @NabeelYousafPasha . Appreciated!

**Pull request closed without merging**